Gems/Multiplayer/Code/Source/Debug/MultiplayerDebugByteReporter.cpp

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "MultiplayerDebugByteReporter.h"
+
+#include <iomanip> // for std::setfill
+#include <sstream>
+#include <AzCore/std/sort.h>
+
+namespace Multiplayer
+{
+    MultiplayerDebugByteReporter::MultiplayerDebugByteReporter()
+    {
+        MultiplayerDebugByteReporter::Reset();
+    }
+
+    void MultiplayerDebugByteReporter::ReportBytes(size_t byteSize)
+    {
+        m_count++;
+        m_totalBytes += byteSize;
+        m_totalBytesThisSecond += byteSize;
+        m_minBytes = AZStd::min(m_minBytes, byteSize);
+        m_maxBytes = AZStd::max(m_maxBytes, byteSize);
+    }
+
+    void MultiplayerDebugByteReporter::AggregateBytes(size_t byteSize)
+    {
+        m_aggregateBytes += byteSize;
+    }
+
+    void MultiplayerDebugByteReporter::ReportAggregateBytes()
+    {
+        ReportBytes(m_aggregateBytes);
+        m_aggregateBytes = 0;
+    }
+
+    float MultiplayerDebugByteReporter::GetAverageBytes() const
+    {
+        if (m_count == 0)
+        {
+            AZ_Warning("MultiplayerDebugByteReporter", m_totalBytes == 0, "Attempted to average bytes with a zero count.");
```

Comment:
This seems like a reasonable edge case, I'm not sure a warning is necessary here.

Reply:
That is fair, removing.